pkg/server/server.go

```diff
@@ -0,0 +1,629 @@
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/labels"
+
+	k8sClient "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+//Server struct
+type Server struct {
+	Router *mux.Router
+}
+
+//Global Kubernetes Client
+var client k8sClient.Client
+
+//Init does stuff
+func Init(clientConfig restclient.Config) error {
+	var err error
+	var tempClient *k8sClient.Client
+
+	//In Cluster Config
+	if clientConfig.Host == "" {
+		tempConfig, err := restclient.InClusterConfig()
+		if err != nil {
+			return err
+		}
+		tempClient, err = k8sClient.New(tempConfig)
+
+		client = *tempClient
+
+		//Local Config
+	} else {
+		tempClient, err = k8sClient.New(&clientConfig)
+		if err != nil {
+			return err
+		}
+		client = *tempClient
+	}
+
+	return nil
+}
+
+//NewServer creates a new server
+func NewServer() (server *Server) {
+	router := mux.NewRouter()
+
+	sub := router.PathPrefix("/beeswax/deploy/api/v1").Subrouter()
+
+	sub.Path("/environmentGroups").Methods("GET").HandlerFunc(getEnvironmentGroups)
+
+	sub.Path("/environmentGroups/{environmentGroupID}").Methods("GET").HandlerFunc(getEnvironmentGroup)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments").Methods("GET").HandlerFunc(getEnvironments)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments").Methods("POST").HandlerFunc(createEnvironment)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}").Methods("GET").HandlerFunc(getEnvironment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}").Methods("PATCH").HandlerFunc(updateEnvironment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}").Methods("DELETE").HandlerFunc(deleteEnvironment)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments").Methods("GET").HandlerFunc(getDeployments)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments").Methods("POST").HandlerFunc(createDeployment)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments/{deployment}").Methods("GET").HandlerFunc(getDeployment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments/{deployment}").Methods("PATCH").HandlerFunc(updateDeployment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments/{deployment}").Methods("DELETE").HandlerFunc(deleteDeployment)
+
+	server = &Server{
+		Router: router,
+	}
+	return server
+}
+
+//Start the server
+func (server *Server) Start() error {
+	return http.ListenAndServe(":9000", server.Router)
+}
+
+//Route handlers
+
+//getEnvironmentGrouos returns a list of all Environment Groups
+//What is an environmentGroup?
+func getEnvironmentGroups(w http.ResponseWriter, r *http.Request) {
+	//TODO: What is this supposed to do?
+	//For now I'll just return 404 I guess...
+	http.Error(w, "404 Page not found", 404)
```

Comment:
If the point of these is to act as a _"Not Implemented"_, you could return a `501` but chances are good this should be a `405` _(Method Not Allowed)_ until you get it implemented because it's not really an active/live endpoint.